cmd/secrets/secret_helper.go

```diff
@@ -53,105 +81,112 @@ type secretsRequest struct {
 	Secrets []string `json:"secrets"`
 }
 
-// ReadSecrets implements a secrets reader from a directory/mount
-func readSecrets(r io.Reader, w io.Writer, dir string) error {
-	in, err := ioutil.ReadAll(r)
+type backend interface {
```

Comment:
nit: as mentioned in an earlier comment, `backend` is ambiguous. While not a big deal for var names, in interfaces, I think it would be very valuable to have a more-specific name (e.g. `secretsProvider` or something similar).

Reply:
nit: Since this interface is not really just a part of the CLI or the helper itself, it may be good to separate it into its own file and/or versioned package.

Reply:
Agreed. Already answered those 2 points in other comments.